tests/data/MerchFactory.ts

```diff
@@ -38,21 +46,35 @@ export class MerchFactory {
     return substitutes.map((sub) => MerchandiseItemOptionModel.merge(MerchFactory.fakeOption(), sub));
   }
 
+  public static optionMetadataWith(...substitutes: Partial<MerchItemOptionMetadata>[]): MerchItemOptionMetadata[] {
+    return substitutes.map((sub) => {
+      const metadata = MerchFactory.fakeOptionMetadata();
+      return {
+        type: sub.type ?? metadata.type,
+        value: sub.value ?? metadata.value,
+        position: sub.position ?? metadata.position,
+      };
+    });
+  }
+
   public static fakeCollection(): MerchandiseCollectionModel {
     return MerchandiseCollectionModel.create({
       uuid: uuid(),
-      title: faker.random.hexaDecimal(10),
+      title: faker.datatype.hexaDecimal(10),
       description: faker.lorem.sentences(2),
       items: MerchFactory.createItems(FactoryUtils.getRandomNumber(1, 5)),
     });
   }
 
   public static fakeItem(): MerchandiseItemModel {
+    const hasVariants = FactoryUtils.getRandomBoolean();
+    const maxNumOptions = hasVariants ? 5 : 1;
     return MerchandiseItemModel.create({
       uuid: uuid(),
-      itemName: faker.random.hexaDecimal(10),
+      itemName: faker.datatype.hexaDecimal(10),
       description: faker.lorem.sentences(2),
-      options: MerchFactory.createOptions(FactoryUtils.getRandomNumber(1, 5)),
+      hasVariants,
+      options: MerchFactory.createOptions(FactoryUtils.getRandomNumber(1, maxNumOptions)),
```

Comment:
Shouldn't this also depend on `hasVariants`? Maybe instead of `maxNumOptions`,
```typescript
const numOptions = hasVariants ? FactoryUtils.getRandomNumber(1, 5) : 1;
```